common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -552,6 +553,76 @@ public ICacheRecord load(@NonNull final String clientId,
                 null // not applicable
         );
 
+        {
+            // If we didn't find an RT in the cache, this could be a "TSL-seed" or "dual-client stack"
+            // scenario
+            //
+            // Defining these terms:
+            // TSL-seed: another 1P TSL integrated app has put a token into our cache so we can
+            //     pick it up
+            //
+            // Dual-Client stack: two FoCI-enabled app registrations are sharing a single binary
+            //     and accordingly, can share RTs.
+            //     Examples for this might be TFL/TFW - which uses multiple client ids to enable
+            //     different scenarios depending on enterprise vs. consumer usage
+
+            // Unlike the broker, where we check if an app is FoCI prior to making a network call
+            // with an arbitrary FoCI RT we find in the cache, if we're in standalone mode and find
+            // a FoCI RT in the cache, the current app must also be FoCI (!!!)
+            //
+            // Making the assumption that the current client id can use any FoCI RT we find in the
+            // cache is strictly contingent that app developers NOT mix 1P & 3P registrations into
+            // the same binary. If you do this, Bad Things will happen and you'll get confusing
+            // errors that the RT used doesn't match the client app registration. Also, this
+            // assumption means we don't need to implement "FoCI probing" and/or track FoCI app meta
```

Comment:
Left similar comment below, basically I want to know if some app can have multiple client id stacked, let's say three, and all of them are 1P registrations. Could there be case that even though all three are 1P, but for some reason one of them is not part of FoCI, whereas the other two are. Is this case possible? And should we implement probing for it?

Reply:
See comment [here](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/893#discussion_r413221883) -- we won't support this by-design. Not supporting it let's us dodge having to bring probe logic out of broker